File: Irrgarten/src/irrgarten/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author Sergio Salvador Gil
 */
public class TestP1 {
    
    public static void main() {
        
        // Índice de pruebas
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");
        
        System.out.println("Pruebas:");
        System.out.println("1-Clase Gamestate\n2-Clase Dice\n3-Clases Weapon y Shield\n4-Structs\n");
        
        // Prueba del GameState
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");
        
        System.out.println("1- Clase GameState:\n");
        
        GameState estado = new GameState("Namibia", "No hay jugadores",
                "No hay monstruos", 0, false, "No ha ocurrido nada");
        
        System.out.println("Labyrinth: " + estado.getLabyrinth());
        System.out.println("Players: " + estado.getPlayers());
        System.out.println("Monsters: " + estado.getMonsters());
        System.out.println("Current Player: " + estado.getCurrentPlayer());
        System.out.println("Winner? " + estado.isWinner());
        System.out.println("Log: " + estado.getLog()+ "\n");
        
        // Prueba de la clase Dice
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");
        System.out.println("2- Clase Dice:\n");
        Dice dado = new Dice();
        
        System.out.println("Posiciones aleatorias [0,9]:");
        int num = 10;
        for(int i = 0; i < num; i++){
            
            if(i==num-1){
                System.out.println(dado.randomPos(10)+"\n");
            }
            else{
                System.out.print(dado.randomPos(10)+", ");
            }
        }
        
        System.out.println("Fuerza de un arma aleatoria (hasta 10):");
        
        for(int i = 0; i < num; i++){
            
            if(i==num-1){
                System.out.println(dado.randomStrength()+"\n");
            }
            else{
                System.out.print(dado.randomStrength()+", ");
            }
        }
        
        // Los métodos con implementaciones idénticas (Que simplemente hagan uso
        // de los métodos nextInt() y nextFloat() de la clase Random) se dan 
        // por probados.
        
        System.out.println("Intentos de resurrección de un jugador:");
        
        for(int i = 0; i < num; i++){
            
            if(i==num-1){
                System.out.println(dado.resurrectPlayer());
            }
            else{
                System.out.print(dado.resurrectPlayer()+", ");
            }
        }
        
        int exitos = 0;
        num=1000000;
        
        for(int i = 0; i < num; i++){
            
            if(dado.resurrectPlayer()){
                exitos++;
            }
        }
        
        float p = exitos/(float)num;
        
        System.out.println("Tras "+num+" intentos, la probabilidad calculada de resurrección es de: "+p+"\n");
        
        System.out.println("Descartar elementos:");
        
        num=10;
        for(int i = 0; i<num; i++){
            
            int usos=dado.usesLeft();
            
            System.out.println("Usos: "+usos+"/5. Descartado: "+ dado.discardElement(usos));
            
        }
        
        // Prueba de las clases Weapon y Shield
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");
        System.out.println("3- Clases Weapon y Shield:\n");
        
        System.out.println("Generación de armas y escudos:");
        for(int i = 0; i<num; i++){
            Weapon w= new Weapon(dado.weaponPower(), dado.usesLeft());
            if(i==num-1){
                System.out.print(w.toString()+"\n");
            }
            else{
                System.out.print(w.toString()+", ");
            }
        }
        
        for(int i = 0; i<num; i++){
            Shield s= new Shield(dado.shieldPower(), dado.usesLeft());
            if(i==num-1){
                System.out.println(s.toString()+"\n");
            }
            else{
                System.out.print(s.toString()+", ");
            }
            
        }
        
        System.out.println("Descartar armas y escudos:");
        
        
        
        //Prueba de los Structs
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");
        System.out.println("4- Structs:\n");
        
        
        // Fin de las pruebas
        System.out.println("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n");


    }
}

File: rb/Irrgarten/test_p1.rb
# encoding: utf-8

require_relative 'GameState'
require_relative 'Dice'
require_relative 'Weapon'
require_relative 'Shield'
require_relative 'Directions'
require_relative 'GameCharacter'
require_relative 'Orientation'

module Irrgarten
  class TestP1
    def self.main
      # Índice de pruebas
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
      
      puts "Pruebas:"
      puts "1-Clase Gamestate\n2-Clase Dice\n3-Clases Weapon y Shield\n4-Structs\n\n"
      
      # Prueba del GameState
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
      
      puts "1- Clase GameState:\n\n"
      
      estado = GameState.new("Namibia", "No hay jugadores", "No hay monstruos", 0, false, "No ha ocurrido nada")
      
      puts "Labyrinth: #{estado.get_labyrinth}"
      puts "Players: #{estado.get_players}"
      puts "Monsters: #{estado.get_monsters}"
      puts "Current Player: #{estado.get_current_player}"
      puts "Winner? #{estado.get_winner}"
      puts "Log: #{estado.get_log}\n\n"
      
      # Prueba de la clase Dice
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
      puts "2- Clase Dice:\n\n"
      dado = Dice.new
      
      puts "Posiciones aleatorias [0,9]:"
      num = 10
      (0...num).each do |i|
        if i == num - 1
          print "#{dado.random_pos(10)}\n\n"
        else
          print "#{dado.random_pos(10)}, "
        end
      end
      
      puts "Fuerza de un arma aleatoria (hasta 10):"
      (0...num).each do |i|
        if i == num - 1
          print "#{dado.random_strength}\n\n"
        else
          print "#{dado.random_strength}, "
        end
      end
      
      puts "Intentos de resurrección de un jugador:"
      (0...num).each do |i|
        if i == num - 1
          print "#{dado.resurrect_player}\n"
        else
          print "#{dado.resurrect_player}, "
        end
      end
      
      exitos = 0
      intentos = 1_000_000
      
      intentos.times do
        if dado.resurrect_player
          exitos += 1
        end
      end
      
      p = exitos.to_f / intentos
      
      puts "Tras #{intentos} intentos, la probabilidad calculada de resurrección es de: #{p}\n\n"
      
      puts "Descartar elementos:"
      
      (0...num).each do |i|
        usos = dado.uses_left
        puts "Usos: #{usos}/5. Descartado: #{dado.discard_element(usos)}"
      end
      
      # Prueba de las clases Weapon y Shield
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
      puts "3- Clases Weapon y Shield:\n\n"
      
      puts "Generación de armas y escudos:"
      (0...num).each do |i|
        w = Weapon.new(dado.weapon_power, dado.uses_left)
        if i == num - 1
          print "#{w.to_s}\n"
        else
          print "#{w.to_s}, "
        end
      end
      
      (0...num).each do |i|
        s = Shield.new(dado.shield_power, dado.uses_left)
        if i == num - 1
          print "#{s.to_s}\n\n"
        else
          print "#{s.to_s}, "
        end
      end
      
      puts "Descartar armas y escudos:\n\n"
      
      # Prueba de los Structs
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
      puts "4- Structs:\n\n"
      
      # Fin de las pruebas
      puts "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n\n"
    end
  end
end

Irrgarten::TestP1.main